web/fido_app/webauthn_routes.py

```diff
@@ -68,6 +68,34 @@ def webauthn_registration_start():
     return options_to_json(registration_options)
 
 
+@app.route('/webauthn/registration/add-start', methods=['POST'])
+def webauthn_registration_add_start():
+    """Starts the webauthn registration process by sending the user a random challenge"""
+
+    # MakeCredentialOptions
+    email = request.form['email']
+
+    ukey = secrets.token_urlsafe(20)
+    display_name = get_display_name(email)
+    
+    registration_options = generate_registration_options(
+        rp_name=RP_NAME,
+        rp_id=RP_ID,
+        user_id=ukey,
+        user_name=email,
+        user_display_name=display_name
+    )
+
+    session['registration'] = {
+        'email': email,
+        'challenge': bytes_to_base64url(registration_options.challenge),
+        'ukey': ukey,
+        'display_name': display_name
+    }
+
+    return options_to_json(registration_options)
```

Comment:
I'm not sure that this route does what we want. Currently, it has a security bug: a malicious API user could POST to this route with someone else's email and add their own credential! Instead of getting the email from the request, I think we should grab the email from the user's session (which means adding an `@login_required` decorator).

Also, this will still lead to the creation of a new user because the user object is created in `verify_registration_credentials`, not in `webauthn_registration_start()`. I'm not sure how we want to deal with not creating a new user always in that method.